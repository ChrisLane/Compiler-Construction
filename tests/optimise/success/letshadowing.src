// This is to test that variable shadowing works. This should evaluate to 4.
main() {
    let x = 1 {
        let x = $x + $x {
            let x = $x + $x {
                print $x;
            };
        };
    };
}
