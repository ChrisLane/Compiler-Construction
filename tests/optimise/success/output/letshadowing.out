
main (  ) { 
    Print ( Const 4 ) 
}

